### PR TITLE
[DEPRECATION] Deprecate ZPAGES

### LINKED ISSUE #2217: [DEPRECATION] Deprecate ZPAGES
To discuss in C++ SIG meeting.

Related:

* https://github.com/open-telemetry/opentelemetry-specification/pull/3577
* https://github.com/open-telemetry/opentelemetry-java/pull/5578
* https://github.com/open-telemetry/opentelemetry-java/issues/5571
* https://github.com/open-telemetry/opentelemetry-specification/pull/3564
* https://github.com/open-telemetry/opentelemetry-dotnet/pull/4604

### PR DESCRIPTION
**Pull request by marcalff**
Fixes #2217

## Changes

Please provide a brief description of the changes here.

* Deprecate WITH_ZPAGES

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by owent — APPROVED** [review]

**Review by esigo — APPROVED** [review]
LGTM
thanks :)

File: DEPRECATED.md [hunk 1]
```diff
@@ -151,7 +151,59 @@ N/A
...
+  * the C++ class `TracezHttpServer`
+  * the C++ class `TracezSpanProcessor`
+  * the C++ class `TracezSharedData`
+  * the C++ class `ZPages`
+  * the C++ class `zPagesHttpServer`
+* all the code and doc located under `ext/src/zpages/`
+* all the tests located under `ext/test/zpages/`
+* the zpages exporter library(`opentelemetry_zpages`)
+* the zpages build options in CMake (`WITH_ZPAGES`)
+
```
Comment (lalitb) [inline DEPRECATED.md:192]: Nit - [HttpServer](ext/include/opentelemetry/ext/http/server) was used by ZPages, apart from examples and test code.  With ZPages removal, we can also modify [cmake](https://github.com/open-telemetry/opentelemetry-cpp/blob/01e65810b9267df34487ce6bd4ed175be6b9d51e/ext/CMakeLists.txt#L21) and [bazel ](https://github.com/open-telemetry/opentelemetry-cpp/blob/01e65810b9267df34487ce6bd4ed175be6b9d51e/ext/BUILD#L8) script to not install HttpServer headers. This server is not production-ready, and shouldn't be used for any other purpose.

File: DEPRECATED.md [hunk 2]
```diff
@@ -151,7 +151,59 @@ N/A
...
+Consider using a different exporter,
+for example the OTLP exporter (both OTLP HTTP and OTLP GRPC are supported),
+to expose trace data to a separate trace backend.
+
+Note that this changes the access pattern:
+
+* with zpages, data is only available locally (in process)
+* with other exporters, data is available externally (not in process)
+
```
Comment (lalitb) [inline DEPRECATED.md:203]: Nit - Probably we can also mention if someone is using ZPages, they can take ownership of this component, and add it to the contrib repo -  similar to how Golang maintains [ZPages ](https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/zpages)implementation.

**Review by lalitb — APPROVED** [review]
LGTM. Thanks.

**Comment by marcalff** [comment]
> LGTM with nit suggestions. Thanks.

Thanks, will adjust patch accordingly.

File: DEPRECATED.md (hunk 1 above)
Comment (marcalff) [reply DEPRECATED.md:192]: Thanks, good point. Fixed.

File: DEPRECATED.md (hunk 2 above)
Comment (marcalff) [reply DEPRECATED.md:216]: Thanks. Fixed.

**Pull request merged**